linux/just_git_airgap_repo.bsh

```diff
@@ -0,0 +1,1507 @@
+# 4. Extract the archive (the archive will extract directly into this directory)
+#   a) In vsi_common_prep, edit repo_map.env and set the JUST_GIT_AIRGAP_MIRROR_URL environment variable. For example,
+#     - JUST_GIT_AIRGAP_MIRROR_URL=https://git-server.com/projectA
+# 5. If not already, initialize bare repositories on the air-gapped git server for vsi_common and its submodule, recipes. In this example, these should be located at
+#   - https://git-server.com/projectA/vsi_common.git
+#   - https://git-server.com/projectA/recipes.git
+# 6. ``source setup.env``
+# 7. ``just git import-repo`` - Push the mirrored repository and all submodules to a new git server as defined by repo_map.env
+# 8. Incremental updates can be pushed to the repositories using much the same process. See :file:`git_mirror_main`.
+#
+# A developer can then run:
+# 1. ``git clone https://git-server.com/projectA/vsi_common.git``
+# 2. ``just git clone-airgap-repo`` - Clone submodules recursively from the new mirror
+#
+# Note however, that a project that uses vsi_common as a submodule must add support in the setup.env file to initialize and update the vsi_common submodule before this target can be called (see :func:`print_git_airgap_submodule_update`).
+#
+# .. seealso::
+#
+#   :file:`git_mirror`: :func:`git_mirror_main`, :func:`git_push_main`, and :func:`git_clone_main`
+#**
+
+#**
+# .. envvar:: GIT
+#
+# Name/path of git executable
+#
+# Instead of hard-coding "git" everywhere, use the variable :envvar:`GIT` so that when the need comes, it is easier to switch which executable gets called. Set to empty string to have any calls to git skipped.
+#
+# All :func:`relocate_git_defaultify` targets will be skipped if the value of :envvar:`GIT` is not found or empty.
+#
+# .. note::
+#
+#   When using the GIT env variable in this file, do not surround it with quotes as you normally would. Although this prevents using a path to ``git`` with a space in it, in exchange, it may be possible (eventually) to enable a dry-run-ish capability if GIT is set to ``echo git``.
+#**
+
+: ${GIT=git}
+
+#**
+# .. function:: log_unpushed_commits
+#
+# List all unpushed commits to stdout
+#
+# List all commits on branches that have not been pushed to a remote. This is a set operation: it lists all the commits from a branch in the DAG except those reachable by a remote-tracking branch. It does not require an *associated* remote-tracking branch
+#
+# REVIEW What order should these go in
+# :Arguments: - [``$1``] - A branch's name to filter by; e.g., master. If unset, then don't filter
+#             - [``$2``] - A name of a remote to filter by; e.g., origin. If unset, then don't filter
+#
+# :Output: - ``*stdout*`` - Print all unpushed commits for tracking branches to stdout
+#**
+
+function log_unpushed_commits()
+{
+  # NOTE Only supports passing in a single branch and named remote
+  local branch="${1---branches}"
```

Comment:
What's wrong with `--all`?

If you do `--branch --tags` remember, then when you use `${branch}` in bash, you have to not use quotes, so that the space is interpreted as two separate arguments.